nixos/modules/services/networking/go-away.nix

```
@@ -0,0 +1,257 @@
+                              if config.tlsCertificate != null then
+                                "/run/credentials/${instanceName name}.service/tlsCertificate"
+                              else
+                                null;
+                            readOnly = true;
+                          };
+                          tls-key = lib.mkOption {
+                            type = types.nullOr types.str;
+                            default =
+                              if config.tlsKey != null then "/run/credentials/${instanceName name}.service/tlsKey" else null;
+                            readOnly = true;
+                          };
+                          # TODO
+                        };
+                      };
+                    };
+                    # TODO
+                  };
+                };
+              };
+
+              policy = lib.mkOption {
+                default = { };
+                description = ''
+                  TODO
+                '';
+                type = types.submodule {
+                  freeformType = yamlFormat.type;
+                  options = {
+                    # TODO
+                  };
+                };
+              };
+            };
+          }
+        )
+      );
+    };
+  };
+
+  config = lib.mkIf (enabledInstances != { }) {
+    environment.etc = lib.mapAttrs' (
+      name: instance:
+      lib.nameValuePair (policyEtcPath name) ({
+        source = policyFile name instance;
+      })
+    ) enabledInstances;
+    systemd.services = lib.mapAttrs' (
+      name: instance:
+      lib.nameValuePair "${instanceName name}" {
+        description = "Go-Away (${if name == "" then "default" else name} instance)";
+        wantedBy = [ "multi-user.target" ];
+        wants = lib.lists.optional (
+          instance.useACMEHost != null
+        ) "acme-finished-${instance.useACMEHost}.target";
+        after = [
+          "network.target"
+        ] ++ (lib.lists.optional (instance.useACMEHost != null) "acme-${instance.useACMEHost}.service");
+        reloadTriggers = [ (policyFile name instance) ];
+
+        serviceConfig = {
+          DynamicUser = true;
+
+          ExecStart = utils.escapeSystemdExecArgs (
+            (lib.singleton (lib.getExe cfg.package))
+            ++ [
+              "-config"
+              (configFile name instance)
+              "-policy"
+              "/etc/${policyEtcPath name}"
+              # TODO Snippets. Would need to include the example things though.
+              "-cache"
+              "/var/cache/${instanceName name}"
+              # TODO Other arguments that don't have a corresponding YAML option: -client-ip-header, -path
+              # TODO Thing that should be handled as credential JWT_PRIVATE_KEY_SEED
+            ]
+            ++ instance.extraFlags
+          );
+          ExecReload = "${pkgs.coreutils}/bin/kill -SIGHUP $MAINPID";
+          CacheDirectory = instanceName name;
```

```suggestion
          CacheDirectory = "go-away/${name}";
```

Nesting like that would be just fine.